src/palindrom.js

```diff
@@ -509,77 +548,53 @@ const Palindrom = (() => {
                 headers['X-Referer'] = this.remoteUrl.pathname;
             }
             if (method === 'GET') {
-                requestPromise = axios.get(url, {
+                responsePromise = axios.get(url, {
                     headers
                 });
             } else {
-                requestPromise = axios.patch(url, data, {
+                responsePromise = axios.patch(url, data, {
                     headers
                 });
             }
-            requestPromise
-                .then(res => {
-                    this.handleSuccessResponse(res, callback);
-                })
-                .catch(error => {
-                    const res = error.response;
 
-                    if (res) {
-                        var statusCode = res.status;
-                        var statusText = res.statusText || res.data;
-                        var reason = res.data;
-
-                        //this is not a fatal error
-                        if (
-                            statusCode >= 400 &&
-                            statusCode < 500 &&
-                            res.headers['content-type'] ===
-                                'application/json-patch+json'
-                        ) {
-                            this.handleSuccessResponse(res, callback);
-                            return;
-                        }
+            this.onSend(data, url, method);
+
+            return new Promise(async (resolve, reject) => {
+                try {
```

Comment:
Do we really need all those try catches everywhere and return new promise? I thought axios returns a promise for the request. A promise that rejects, so you can use `.catch` to transform rejected 4xx promise to resolved one and forward rejection otherwise.
So the code would look more like:
```js
return responsePromise.catch(error, (error)=>{
  if(errorIs4xx(error)){
    return error.response;
  }
  return Promise.reject(error);
});
```

I think this reduces few lines of `new Promise`, `async ()=>{}`, `try{}catch(){}` and does exactly same thing from the perspective of `async xhr` API user